app/controllers/incomeSources/add/CheckBusinessDetailsController.scala

```diff
@@ -180,4 +175,57 @@ class CheckBusinessDetailsController @Inject()(val checkBusinessDetails: CheckBu
         }
   }
 
+  private def getBusinessDetailsFromSession(implicit user: MtdItUser[_], ec: ExecutionContext): Future[Either[Throwable, CheckBusinessDetailsViewModel]] = {
+    val userActiveBusinesses: List[BusinessDetailsModel] = user.incomeSources.businesses.filterNot(_.isCeased)
+    val skipAccountingMethod: Boolean = userActiveBusinesses.isEmpty
+
+    // List of fields we are attempting to extract
+    val fields: Seq[String] = Seq(SessionKeys.businessName, SessionKeys.businessStartDate, SessionKeys.businessTrade,
+      SessionKeys.addBusinessAddressLine1, SessionKeys.addIncomeSourcesAccountingMethod, SessionKeys.addBusinessAccountingPeriodEndDate, SessionKeys.addBusinessAddressLine2,
+      SessionKeys.addBusinessAddressLine3, SessionKeys.addBusinessAddressLine4, SessionKeys.addBusinessPostalCode, SessionKeys.addBusinessCountryCode, SessionKeys.addIncomeSourcesAccountingMethod)
+
+    val init: Future[Either[Throwable, Map[String, Option[String]]]] = Future {
+      Right(Map.empty)
+    }
+    val xs = fields.foldLeft(init) { (acc, fieldName) =>
+      for {
+        futureAcc <- acc
+        if futureAcc.isRight // stop exec if there was any error previously
+        futureRes <- sessionService.get(fieldName)
+      } yield (futureAcc, futureRes) match {
+        case (Right(m), Right(optVal)) =>
+          val finalMap = m + (fieldName -> optVal)
+          Right(finalMap)
+        case (_, _) =>
+          futureAcc
+      }
+    }
+
+    for {
+      res <- xs
+    } yield res match { // => Add extra validation for the date parsing + others where required
```

Comment:
I was hoping we would go data validation in each of fields we calling .get when constructing model, but assuming this is a temp code and we would use something provided by relevant service, we might skip this on this occasion